### PR TITLE
Revising app config and schedule plan deletion behavior based on feedback, adding more tests.

### PR DESCRIPTION
**Pull request by alxdarksage**
Continuing to document the overall behavior each object should exhibit through the APIs here:

https://sagebionetworks.jira.com/wiki/spaces/BRIDGE/pages/585072641/Logical+Physical+Delete+Behavior+for+model+objects

Document now includes a list of tests

### CODE REVIEW

File: app/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDao.java [hunk 1]
```diff
@@ -116,24 +117,24 @@ public void deleteAppConfig(StudyIdentifier studyId, String guid) {
...
         checkNotNull(guid);
         
-        // If app config doesn't exist, fail silently. It's more user-friendly
         DynamoAppConfig appConfig = loadAppConfig(studyId, guid);
-        if (appConfig != null) {
-            mapper.delete(appConfig);
-            criteriaDao.deleteCriteria(appConfig.getCriteria().getKey());
+        if (appConfig == null) {
+            throw new EntityNotFoundException(AppConfig.class);
         }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDao.java:135]: Unit test for delete permanently a missing app config?

File: test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDaoTest.java [hunk 2]
```diff
@@ -143,7 +144,7 @@ public void logicalDeleteLeavesConfigInDatabase() {
     }
     
     @Test
-    public void cannotCreateConfigInDeletedState() {
+    public void createAppleCannotBeDeleted() {
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDaoTest.java]: Nitpick: Typo "Apple"

File: app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java [hunk 3]
```diff
@@ -108,7 +108,11 @@ public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, Schedule
     public SchedulePlan updateSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
         checkNotNull(studyIdentifier);
         checkNotNull(plan);
-        
+
+        SchedulePlan saved = getSchedulePlan(studyIdentifier, plan.getGuid());
+        if (saved == null || (saved.isDeleted() && plan.isDeleted())) {
+            throw new EntityNotFoundException(SchedulePlan.class);
+        }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java:115]: Unit test for updating a missing schedule plan?

File: test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java [hunk 4]
```diff
@@ -180,36 +190,43 @@ public void createSchedulePlanWritesCriteria() {
...
-        newCriteria.setKey(crit.getKey());
-        newCriteria.setMinAppVersion(IOS, 100);
-        newCriteria.setMaxAppVersion(IOS, 200);
-        strategy.getScheduleCriteria().set(0, new ScheduleCriteria(scheduleCriteria.getSchedule(), newCriteria));
+        SchedulePlan updated = dao.updateSchedulePlan(TEST_STUDY, update);
         
-        plan = dao.updateSchedulePlan(TEST_STUDY, plan);
+        // Verify returned object still has changes
+        strategy = (CriteriaScheduleStrategy)updated.getStrategy();
+        strategy.getScheduleCriteria().get(0).getCriteria();
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java]: Nitpick: Dangling code line.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
Mostly good. A few test notes and a few nitpicks.

File: app/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDao.java (hunk 1 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDao.java:135]: Added

File: test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDaoTest.java (hunk 2 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDaoTest.java]: changed

File: app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java (hunk 3 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java:115]: Added two tests to test EntityNotFoundException is thrown for missing schedule plan and for deleted schedule plan when the provided update is also deleted.

File: test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java (hunk 4 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java]: removed

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**